orchestrator/services/celery.py

```diff
@@ -39,17 +44,20 @@ def _celery_start_process(
     task_name = NEW_TASK if workflow.target == Target.SYSTEM else NEW_WORKFLOW
     trigger_task = get_celery_task(task_name)
     pstat = create_process(workflow_key, user_inputs, user)
-
     tasks = pstat.state.s
-    result = trigger_task.delay(pstat.process_id, workflow_key, tasks, user)
-
-    # Enables "Sync celery tasks. This will let the app wait until celery completes"
-    if app_settings.TESTING:
-        process_id = result.get()
-        if not process_id:
-            raise RuntimeError("Celery worker has failed to resume process")
-
-    return pstat.process_id
+    try:
+        result = trigger_task.delay(pstat.process_id, workflow_key, tasks, user)
+        # Enables "Sync celery tasks. This will let the app wait until celery completes"
+        if app_settings.TESTING:
+            process_id = result.get()
+            if not process_id:
+                raise RuntimeError("Celery worker has failed to resume process")
+        return pstat.process_id
+    except (ConnectionError, OperationalError) as e:
+        # If connection to Redis fails and process can't be started, we need to remove the created process
+        logger.warning("Celery worker connection error")
+        delete_process(pstat.process_id)
```

Comment:
Can you elaborate?
I think it should only remove the process that was created by the `create_process` call on line 46. It creates a new process_id

Reply:
yea, I meant that it would delete the created process from line 46 that is a process with user input, which I believe we don't want when only the redis connection fails, I believe you want to retry it later when the connection works again.

Reply:
I see. The `trigger_task.delay` call has a retry built-in (20x retry with 1s interval) and is blocking until the message has been delivered to Celery.

Only when the Redis is completely down and fails does the call return to the frontend with an Internal Server error, but the frontend would have been waiting for 20s already.

We could store the user input somewhere intermediate, but that's a bigger change. It probably shouldn't happen often enough for the lost user-input to be a problem...

Reply:
The `Created` status should be good enough place to keep the process with user_input data

Reply:
I'dd say that the moment we do a `create_process` which saves the process to the DB, that frontend shouldn't be involved anymore.

Reply:
This is called from the `POST /processes/<workflow_key>` endpoint which needs to give a quick response, it can't wait forever for Redis to be back.

Reply:
Saving the process in the CREATED state was what it already did before, but when Celery can't process it, it is stuck. If it can't be handled right away, when do we retry again and what background process will be responsible?

Sounds like that would require another process state, like 'STARTED`.
- `CREATED` (saved in the api-server)
- `STARTED` (picked up in the worker)
- `RUNNING` (entered first step)

Would be a more involved solution for sure